### PR TITLE
New command line utility tool to print planning scene info

### PR DESCRIPTION
**Pull request by davetcoleman**
There was already a ``moveit_print_planning_model_info`` utility so I copied that and adapted it as a debugging tool for the planning_scene attached objects

### CODE REVIEW

**Comment by davetcoleman** [comment]
Added to tutorials: https://github.com/ros-planning/moveit_tutorials/pull/263

File: moveit_ros/planning/planning_components_tools/src/print_planning_scene_info.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+  // Create planning scene monitor
+  planning_scene_monitor::PlanningSceneMonitor psm(ROBOT_DESCRIPTION, tf_buffer);
+  if (!psm.getPlanningScene())
+  {
+    ROS_ERROR_STREAM_NAMED(LOGNAME, "Planning scene not configured");
+    return 1;
+  }
+
+  // TODO(davetcoleman): Support namepacing of the move_group planning_scene service
```
Comment (v4hn) [inline moveit_ros/planning/planning_components_tools/src/print_planning_scene_info.cpp]: I would argue this is supported already.
The tool can be started with a matching namespace.

File: moveit_ros/planning/planning_components_tools/src/print_planning_scene_info.cpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  psm.requestPlanningSceneState(planning_scene_monitor::PlanningSceneMonitor::DEFAULT_PLANNING_SCENE_SERVICE);
+
+  ros::Duration(1.0).sleep();
```
Comment (v4hn) [inline moveit_ros/planning/planning_components_tools/src/print_planning_scene_info.cpp]: The `sleep` appears to be superfluous.
`requestPlanningSceneState` calls the planning scene service and receives the current scene from there

File: moveit_ros/planning/planning_components_tools/src/print_planning_scene_info.cpp [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  ros::AsyncSpinner spinner(1);
+  spinner.start();
+
+  ros::NodeHandle nh;
+  ROS_INFO_STREAM_NAMED(LOGNAME, "Getting planning scene info to print");
+
+  // Create tf transform buffer and listener
+  std::shared_ptr<tf2_ros::Buffer> tf_buffer = std::make_shared<tf2_ros::Buffer>();
+  std::shared_ptr<tf2_ros::TransformListener> tf_listener =
+      std::make_shared<tf2_ros::TransformListener>(*tf_buffer, nh);
```
Comment (v4hn) [inline moveit_ros/planning/planning_components_tools/src/print_planning_scene_info.cpp]: I believe there's no need for the `tf_listener` in this use-case (it's only required when you start topic monitors).

File: moveit_ros/planning/planning_components_tools/src/print_planning_scene_info.cpp [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  {
+    ROS_ERROR_STREAM_NAMED(LOGNAME, "Planning scene not configured");
+    return 1;
+  }
+
+  psm.requestPlanningSceneState(planning_scene_monitor::PlanningSceneMonitor::DEFAULT_PLANNING_SCENE_SERVICE);
+
+  psm.getPlanningScene()->printKnownObjects();
+
+  ros::shutdown();
```
Comment (v4hn) [inline moveit_ros/planning/planning_components_tools/src/print_planning_scene_info.cpp]: This function is always triggered by ROS on-exit and thus does not need to be called here.

File: moveit_ros/planning/planning_components_tools/src/print_planning_scene_info.cpp [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  ros::init(argc, argv, "print_model_info_to_console");
+
+  ros::AsyncSpinner spinner(1);
+  spinner.start();
+
+  ros::NodeHandle nh;
+  ROS_INFO_STREAM_NAMED(LOGNAME, "Getting planning scene info to print");
+
+  // Create tf transform buffer and listener
+  std::shared_ptr<tf2_ros::Buffer> tf_buffer = std::make_shared<tf2_ros::Buffer>();
```
Comment (v4hn) [inline moveit_ros/planning/planning_components_tools/src/print_planning_scene_info.cpp]: The comment is outdated now :)
But why are you still creating a tf2 buffer?
The `PlanningSceneMonitor` provides a constructor with needs only `ROBOT_DESCRIPTION`.
That should suffice here.

File: moveit_ros/planning/planning_components_tools/src/print_planning_scene_info.cpp [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+   Desc:   Simple utility to see all the collision objects in a planning scene, including attached
+*/
+
+#include <ros/ros.h>
+
+// MoveIt!
+#include <moveit/planning_scene_monitor/planning_scene_monitor.h>
+
+// Transforms
+#include <tf2_ros/transform_listener.h>
```
Comment (v4hn) [inline moveit_ros/planning/planning_components_tools/src/print_planning_scene_info.cpp]: Should not be needed anymore (see below).

**Comment by davetcoleman** [comment]
Fixed @v4hn's comments

File: moveit_ros/planning_interface/test/python_move_group.test [hunk 7]
```diff
@@ -1,5 +1,9 @@
+<!-- -*- mode: XML -*- -->
```
Comment (v4hn) [inline moveit_ros/planning_interface/test/python_move_group.test]: I'll merge this as is, but what is up with this header?
The standard header for xml is `<?xml version="1.0"?>`.
Does your IDE not recognize this one or did I miss a change in preferred format?

**Comment by v4hn** [comment]
Merged.

@davetcoleman I suppose there is no need to backport to kinetic, right?
This would change command line output, although not necessarily in a problematic way.

File: moveit_ros/planning_interface/test/python_move_group.test (hunk 7 above)
Comment (davetcoleman) [reply moveit_ros/planning_interface/test/python_move_group.test]: I've used this for a long time and emacs does properly format the file when its included. Not sure where it came from though, but its in my templates

Comment (rhaschke) [reply moveit_ros/planning_interface/test/python_move_group.test]: That's just a header for emacs to "announce" the file mode for emacs. Not related to XML.

**Comment by v4hn** [comment]
> +
> 
> That's just a header for emacs to "announce" the file mode for emacs. Not related to XML.

Yes, but `<?xml version="1.0"?>` tags the format in the same way (verified with emacs)
and is part of the xml standard. It even works with VIM. ;)

@davetcoleman please check your templates again.

**Comment by davetcoleman** [comment]
I had bundled the fix in another PR already, but it looks like that won't get merged https://github.com/ros-planning/moveit/pull/1157

New PR: https://github.com/ros-planning/moveit/pull/1289

**Pull request merged**